mutable_tree.go

```diff
@@ -331,7 +331,10 @@ func (tree *MutableTree) LoadVersionForOverwriting(targetVersion int64) (int64,
 	if err != nil {
 		return latestVersion, err
 	}
-	tree.deleteVersionsFrom(targetVersion + 1)
+	err = tree.deleteVersionsFrom(targetVersion + 1)
+	if err != nil {
+		panic(err)
```

Comment:
we should not be panicing on this error

Reply:
I went for panics because this was never checked before, so the assumption was always it wouldn't error. what would you recommend?

Reply:
I think ignoring the error is the correct behavior in this case. It's possible that a previous version got manually deleted before this function got called. In fact, I don't think it is correct for `deleteVersionsFrom` to return an error in the first place if there happens to be a version in its range that already got deleted.

Reply:
Maybe not panic, but log the error would be useful or we just remove the returning error if we see its not used elsewhere.

Reply:
Just my 2 cents, it's generally advisable that libraries should _not_ panic is almost any circumstance, but rather turn (typed) errors. Let the upstream caller decide what to do with an error and don't assume.